Added option to provide filename

karanmagdani1 opened the pull request:
### Why we need this PR
- Currently, running popeye as a CronJob on K8s saves the filename as `sanitizer_n/a_<timestamp>.txt`
This is due to an [issue](https://github.com/kubernetes/kubernetes/issues/44954) in Kubernetes where API Server does not return the cluster name. 

### What does this PR do
This PR adds an option to provide the filename like shown below:
```
go run main.go --k8s-popeye-cluster-name abcd
Sanitizer saved to: sanitizer_abcd_1581462176820331000.txt
```

- Also added log message to AWS config in case of failure, as it currently does not show any reason.
```
go run main.go --s3-bucket xxxx --k8s-popeye-cluster-name abcd
Sanitizer saved to: sanitizer_abcd_1581462408819395000.txt
2020/02/11 15:06:51 DEBUG: Validate Request s3/PutObject failed, not retrying, error MissingRegion: could not find region configuration
2020/02/11 15:06:51 DEBUG: Build Request s3/PutObject failed, not retrying, error MissingRegion: could not find region configuration
2020/02/11 15:06:51 DEBUG: Sign Request s3/PutObject failed, not retrying, error MissingRegion: could not find region configuration
exit status 1
```

### Further Considerations
I am also considering making the same change [here](https://github.com/derailed/popeye/blob/master/internal/k8s/client.go#L110) as it currently displays the first line of report as `GENERAL [N/A]`. Please let me know what you think 

Signed-off-by: Karan Magdani <karan.magdani@gmail.com>

guusvw commented on pkg/popeye.go:
```diff
@@ -117,7 +120,8 @@ func (p *Popeye) Sanitize() error {
 			// Create a single AWS session (we can re use this if we're uploading many files)
-			s, err := session.NewSession(&aws.Config{})
+			s, err := session.NewSession(&aws.Config{
+				LogLevel: aws.LogLevel(aws.LogDebugWithRequestErrors)})
```
🎉 👍

derailed commented:
@karanmagdani1 @guusvw So I am wondering if we should actually use something like --popeye-cluster option to the cli instead. This could be used in a wider range of use cases when running popeye in cluster and no kubeconfig is available? That way we can solve the cluster name and file issue in both the report and the reports vs just addressing the file name when running in cluster? 

What do you guys think?

karanmagdani1 commented:
@derailed I completely agree with you. We can address a wide range of possible future issues with your suggestion.

 I will rename the flag to something like `--k8s-popeye-cluster` and use the value for both the filename and the report.

karanmagdani1 commented:
Hello guys! I have updated the name of the flag and added updated the name on the report to use this flag if it exists. Please feel free to review whenever you're free @derailed

derailed commented:
@karanmagdani1 @guusvw Thank you both the weighting in! @karanmagdani1 Thank you for making these changes. I think these look good!
